extensions/wikia/CoppaTool/RenameIPProcess.class.php

```diff
@@ -0,0 +1,427 @@
+			$this->addError( wfMessage( 'coppatool-error-invalid-ip' )->escaped() );
+			return;
+		}
+
+		$wgOldUser = $wgUser;
+		$wgUser = User::newFromName( Wikia::USER );
+
+		$cityDb = WikiFactory::IDtoDB( $wgCityId );
+		$this->addLog( "Processing wiki database: {$cityDb}." );
+
+		$dbw = wfGetDB( DB_MASTER );
+		$dbw->begin();
+		$tasks = self::$mLocalIpDefaults;
+
+		$hookName = 'RenameIP::LocalIP';
+		$this->addLog( "Broadcasting hook: {$hookName}" );
+		Hooks::run( $hookName, [ $dbw, $this->mOldIPAddress, $this->mNewIPAddress, $this, $wgCityId, &$tasks ] );
+
+		foreach ( $tasks as $task ) {
+			$this->addLog( "Updating wiki \"{$cityDb}\": {$task['table']}:{$task['username_column']}" );
+			$this->renameInTable( $dbw, $task['table'], $this->mOldIPAddress, $this->mNewIPAddress, $task );
+		}
+
+		$hookName = 'RenameIP::AfterLocalIP';
+		$this->addLog( "Broadcasting hook: {$hookName}" );
+		Hooks::run( $hookName, [ $dbw, $this->mOldIPAddress, $this->mNewIPAddress, $this, $wgCityId, &$tasks ] );
+
+		$dbw->commit();
+
+		$this->addLog( "Finished updating wiki database: {$cityDb}" );
+
+		$this->addMainLog( "log", RenameIPLogFormatter::wiki( $this->mRequestorName, $this->mOldIPAddress, $this->mNewIPAddress, $wgCityId, $this->mReason,
+			!empty( $this->warnings ) || !empty( $this->errors ) ) );
+
+		$wgUser = $wgOldUser;
+	}
+
+	/**
+	 * @author: Władysław Bodzek
+	 * Really performs a rename task specified in arguments
+	 *
+	 * @param DatabaseBase $dbw Database to operate on
+	 * @param string $table Table name
+	 * @param string $oldusername Old IP
+	 * @param string $newusername New IP
+	 * @param array $extra Extra options (currently: userid_column, username_column, conds)
+	 *
+	 * @return bool
+	 */
+	public function renameInTable( $dbw, $table, $oldIP, $newIP, $extra ) {
+		$dbName = $dbw->getDBname();
+		$this->addLog( "Processing {$dbName}.{$table}.{$extra['username_column']}." );
+
+		try {
+			if ( !$dbw->tableExists( $table ) ) {
+				$this->addLog( "Table \"$table\" does not exist in database {$dbName}" );
+				$this->addWarning( wfMessage( 'coppatool-warn-table-missing', $dbName, $table )->inContentLanguage()->text() );
+
+				return false;
+			}
+
+			$values = array(
+				$extra['username_column'] => $newIP,
+			);
+
+			$conds = array(
+				$extra['username_column'] => $oldIP,
+			);
+
+			if ( !empty( $extra['conds'] ) ) {
+				$conds = array_merge( $extra['conds'], $conds );
+			}
+
+			$opts = array(
+				'LIMIT' => self::MAX_ROWS_PER_QUERY,
+			);
+
+			$affectedRows = 1;
+
+			while ( $affectedRows > 0 ) {
+				$dbw->update( $table, $values, $conds, __CLASS__ . '::' . __METHOD__, $opts );
+				$affectedRows = $dbw->affectedRows();
+				$this->addLog( "SQL: " . $dbw->lastQuery() );
+				$dbw->commit();
+				$this->addLog( "In {$dbName}.{$table}.{$extra['username_column']} {$affectedRows} row(s) was(were) updated." );
+				sleep( RENAMEIP_LOOP_PAUSE );
```

Comment:
Where is this const defined?

Reply:
![Sleep](https://media1.giphy.com/media/xT8qBvH1pAhtfSx52U/giphy.gif)

Reply:
It makes more sense to call `wfWaitForSlaves` here then to sleep...